skcosmo/utils/_pcovr_utils.py

```diff
@@ -59,6 +114,36 @@ def check_lr_fit(regressor, X, y=None):
     return fitted_regressor
 
 
+def check_krr_fit(regressor, K, y):
+    r"""
+    Checks that a (kernel ridge) regressor is fitted, and if not,
+    fits it with the provided data
+
+    :param regressor: sklearn-style regressor
+    :type regressor: object
+    :param X: feature matrix with which to fit the regressor
+        if it is not already fitted
+    :type X: array
+    :param y: target values with which to fit the regressor
+        if it is not already fitted
+    :type y: array
+    """
+    try:
+        check_is_fitted(regressor)
+        fitted_regressor = deepcopy(regressor)
+        _check_dual_coefs(regressor, K, y)
+
+    except NotFittedError:
+        fitted_regressor = clone(regressor)
+
+        # Use a precomputed kernel
+        # to avoid re-computing K
+        fitted_regressor.set_params(kernel="precomputed")
+        fitted_regressor.fit(K, y=y)
```

Comment:
So the non-fit behavior is _not_ whatever was defined in the KRR, but the KPCovR? You need to be more explicit about this or change the default to have the same kernel parameters as the KPCovR (my preference)

Reply:
`K` will have already inherited all of the necessary kernel parameters and be constructed accordingly, so to avoid recomputing it the precomputed kernel is used. The regressor parameters are reset after the fitting. This was something I didn't really like, but it was a necessary evil to avoid recomputing the kernel. This is again obsolete if we include the kernel parameters in the KernelPCovR init

Reply:
On second thought, this is still needed even if we include the kernel parameters in the KPCovR init -- so we have to choose whether we want to (a) have the fitted regressor always use a precomputed kernel, (b) build the kernel twice, or (c) use a precomputed kernel for the regressor and then reset its kernel parameters so that it is fully consistent with the KPCovR arguments (this is what is currently done)